source-mysql/replication.go

```diff
@@ -387,7 +405,18 @@ func (rs *mysqlReplicationStream) run(ctx context.Context, startCursor mysql.Pos
 		case *replication.PreviousGTIDsEvent:
 			logrus.WithField("gtids", data.GTIDSets).Trace("PreviousGTIDs Event")
 		case *replication.QueryEvent:
-			if err := rs.handleQuery(ctx, string(data.Schema), string(data.Query)); err != nil {
+			if string(data.Query) == "COMMIT" {
```

Comment:
Do we need to handle possible `ROLLBACK` queries?

This line from MySQL docs got me worried:

> Every transaction (including [autocommit](https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_autocommit) transactions) is recorded in the binary log as though it starts with a [BEGIN](https://dev.mysql.com/doc/refman/5.7/en/commit.html) statement, and ends with either a [COMMIT](https://dev.mysql.com/doc/refman/5.7/en/commit.html) or a [ROLLBACK](https://dev.mysql.com/doc/refman/5.7/en/commit.html) statement. This is even true for statements affecting tables that use a nontransactional storage engine (such as [MyISAM](https://dev.mysql.com/doc/refman/5.7/en/myisam-storage-engine.html)).

https://dev.mysql.com/doc/refman/5.7/en/replication-features-transactions.html

Reply:
There is no such thing as a `ROLLBACK` query modifying a MyISAM table for the same reason there's no such thing as transactional changes to a MyISAM table.

I can see how the phrasing in the docs might be confusing. A clearer way of restating what they're saying might be:
1. Every change is recorded in the binlog as a full transaction and ends with either a commit or a rollback.
2. Even when you're using MyISAM, each change operation is recorded as though it ends in a commit even though MyISAM doesn't have any concept of commits or rollbacks.